README.md

```diff
@@ -1,246 +1,337 @@
 configuration.
 
-## Create place to store SAML config files and this project
-```
+## Setup
+Manual setups steps to do ahead of time to make the rest of this go a lot smoother.
+
+### Request your IdP Metadata
+Skip to setp [Get your IdP Provided metadata](#Get your IdP Provided metadata) and request your IdP metadata now so you will have it by the time you need it.
+
+### Log into first master and SUDO to root
+All of this will be done on your first OpenShift master. While doing work directly on an OpenShift master is typically discouraged, you need access to files that live on the first master to complete this procedure, you will also need to be root, or be able to sudo to root, to access the required files.
+
+### Set up environment variables
+
+Setting these now will make running future steps much more of just a copy/paste exersize rather then more manual fill in the blank.
+
+```sh
 SAML_CONFIG_DIR=~/saml-config
-mkdir $SAML_CONFIG_DIR
-pushd $SAML_CONFIG_DIR
+SAML_UTILITY_PROJECTS_DIR=~/saml-utility-projects
+SAML_PROXY_FQDN=saml-proxy.CHANGE.ME
+SAML_OCP_PROJECT=ocp-saml-proxy
+OPENSHIFT_MASTER_PUBLIC_URL=
+ENTITY_ID=https://${SAML_PROXY_FQDN}/mellon/metadata
+ENDPOINT_URL=https://${SAML_PROXY_FQDN}/mellon
+```
+* `SAML_CONFIG_DIR` - directory to store all of your SAML configuration
+* `UPSTREAM_PROJECTS_DIR` - directory to check out required upstream projects
+* `SAML_PROXY_FQDN` - This will be the FQDN to your SAML proxy (Apache mod_auth_mellon).
+                      This will typically be something like `saml-proxy.MY-OCP-WILDCARD-DOMAIN`.
+                      If you have `*.apps.non-prod.example.com` as your wildcard domain for your OpenShift cluster then this value would be `saml-proxy.apps.non-prod.example.com`.
+                      You could also create a vanity DNS entry and have it route to your OpenShift routers but that isn't needed or typical unless you don't have the a wildcard DNS entry for your OpenShift cluster.
+* `SAML_OCP_PROJECT` - OpenShift project to store the SAML Proxy resources.
+* `OPENSHIFT_MASTER_PUBLIC_URL` - OpenShift masters public URL. This is the URL you access the OpenShift console on. If using a port other then 443 then include `:PORT` as part of the URL.
+* `ENTITY_ID` - An ID unique to your IdP. By convention this should resolve to your meta data file. In the case of the Apache Mellon container created by this project that would be `https://SAML_PROXY_FQDN/mellon/metadata`
+* `ENDPOINT_URL` - The end point of the Apache Mellon service which based on the template created in this project would be  `https://SAML_PROXY_FQDN/mellon`
+
+## Create place to store SAML config files and clone required utitility projects
+```sh
+mkdir ${SAML_CONFIG_DIR}
+mkdir ${SAML_UTILITY_PROJECTS_DIR}
+pushd ${SAML_UTILITY_PROJECTS_DIR}
 git clone https://github.com/openshift/request-header-saml-service-provider.git
+git clone https://github.com/Uninett/mod_auth_mellon.git
```

Comment:
@brenton this is where that other repo is cloned to get access to https://github.com/Uninett/mod_auth_mellon/blob/master/mellon_create_metadata.sh